source/data-handling-bigfiles-R.Rmd

```diff
@@ -0,0 +1,245 @@
+    
+```{r forloop, message=FALSE, warning=FALSE}
+serial_id_list <- c(853, 860, 783)
+print("Average altitude between 2014-03-01 and 2014-07-01:")
+for (serialid in serial_id_list) {
+    res <- dbSendQuery(conn = db,
+                       sprintf("SELECT date_time, latitude, longitude, altitude
+                                FROM processed_logs 
+                                WHERE device_info_serial = %d 
+                                    AND date_time < '2014-07-01' 
+                                    AND date_time > '2014-03-01'", serialid))
+    data <- dbFetch(res)    
+    print(sprintf("serialID %d: %f", serialid, mean(data$altitude)))
+    }
+```
+
+Remark that we use the `sprintf` function to replace dynamically the serialID in the sqlite query we will execute. For each loop, the `%d` is replaced by the value of the serial id of the respective loop. Read the [manual](https://stat.ethz.ch/R-manual/R-devel/library/base/html/sprintf.html) of the `sprintf` function for more insight and options with this kind of replacement (or substitution).
+
+## Interacting with SQLite databases using dplyr
+
+If you're not comfortable with writing queries in SQL, R has a great alternative: `dplyr`. Dplyr can connect to a SQLite database and you can perform the same operations on it that you would do on a dataframe. However, dplyr will translate your commands to SQL, allowing you to take advantage of the indexes in the SQLite database.
+
+```{r dplyr, message=FALSE, warning=FALSE}
+library(dplyr)
+my_db <- src_sqlite(db.name, create = F)
+bird_tracking <- tbl(my_db, "processed_logs")
+results <- bird_tracking %>%
+    filter(device_info_serial == 860) %>%
+    select(date_time, latitude, longitude, altitude)
+head(results)
+```
+
+Dplyr provides the ability to perform queries as above without the need to know SQL. If you want to learn more about how to use `dplyr` with a SQLite database, head over to the [vignette](https://cran.r-project.org/web/packages/dplyr/vignettes/databases.html).
+
+# Takeaway messages
+
+* If you can comfortably work with the entire dataset in memory, but reading the file is rather slow, consider using the `data.table` package and read the file with its `fread` function.
+* If you need to limit the amount of data stored in memory, we would suggest two options:
+    * If you're stuck to a csv, [use `sqldf`](#Limiting-both-the-number-of-rows-and-the-number-of-columns-using-sqldf)
+    * If you can, use a SQLite database and query it using either [SQL queries](#Working-with-SQLite-databases) or [`dplyr`](#Interacting-with-SQLite-databases-using-dplyr)
+
+
```

Comment:
I think I'd try to read the file with `read.csv` and then use `dbWriteTable`. If the file is too large then you can read it in chunks by setting `skip` and `n.row` and repeat those steps until you reach the end of the file.